packages/components/src/List/ListComposition/Manager/ListManager.component.js

```diff
@@ -21,7 +31,15 @@ function Manager({
 
 	const [displayMode, setDisplayMode] = useState(initialDisplayMode || displayModesOptions[0]);
 	const [columns, setColumns] = useState([]);
-	const [visibleColumns, setVisibleColumns] = useState(initialVisibleColumns);
+
+	const [visibleColumns, setVisibleColumns] = useState(
+		visibleColumnsStorageManager.get(visibleColumnsStorageKey) || initialVisibleColumns,
```

Comment:
Wondering if that's actually what we want, because that would mean at some point `initialVisibleColumns` prop becomes uneffective and parent has no way to regain control over the component (local storage being internally managed) 💭
I haven't something better to propose for now though

Reply:
You're right, we have to fix this point before. 
Because if we add a new column (in the next release), this new column won't display if user has modified visibility.